### PR TITLE
OSOE-564: Simplify timeout configuration in reusable workflows in Lombiq.GitHub.Actions

### PR DESCRIPTION
**Pull request by Piedone**
[OSOE-564](https://lombiq.atlassian.net/browse/OSOE-564)

[OSOE-564]: https://lombiq.atlassian.net/browse/OSOE-564?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@ jobs:
   build-and-test-larger-runners:
     if: github.ref_name != github.event.repository.default_branch
     name: Build and Test - root solution (larger runners)
-    uses: Lombiq/GitHub-Actions/.github/workflows/build-and-test-orchard-core.yml@dev
+    uses: Lombiq/GitHub-Actions/.github/workflows/build-and-test-orchard-core.yml@issue/OSOE-564
```
Comment (Piedone) [inline .github/workflows/build-and-test.yml]: @DJdisco07 be sure to revert this and the below change to `dev` before merging this PR.

**Comment by Piedone** [comment]
@DJdisco07 you'll also need to update the submodule to the merged `dev` branch here.

**Comment by dorraj07** [comment]
@Piedone I don't see changes of the branch reference in the submodules or is it something else ?

**Comment by Piedone** [comment]
Do this, then commit-push OSOCE too: 
image

**Comment by dorraj07** [comment]
the word 'here' at the end gave me a lot of hypotheses on what you mean haha

**Comment by Piedone** [comment]
I hope it's clear now :).

**Review by dorraj07 — APPROVED** [review]

**Pull request merged**